doc/mermaid/failure_based_routing.mermaid

```diff
@@ -0,0 +1,40 @@
+sequenceDiagram
+    participant C as Client
+    participant QE as QueryEngine[pod1]
+    participant FD1 as FiloDB[pod1]
+    participant QE2 as QueryEngine[pod2]
+    participant MP as def materializePlan
+    participant FP as trait FailureProvider
+    participant RP as trait RoutingPlanner
+    C->>QE: Paser.queryToLogicalPlan
+    QE-->>C: LogicalPlan
+    C->>QE: materializePlan
+    QE->>MP: materializePlan
+    MP->>FP: getFailures
+    FP-->>MP: Seq[FailureTimeRange]
+    MP->>RP: plan(Seq[FailureTimeRange])
+    RP-->>MP: Seq[Route]
+    loop for each Route
+        alt is LocalRoute
+            MP ->> MP: updateLogicalPlanWithTimeRange
+            MP ->> MP: walkLogicalPlanTree
+            MP ->> MP: lhs: ExecPlan
+        else is RemoteRoute
+            MP ->> MP: updateLogicalPlanWithTimeRange
+            MP ->> MP: createRemoteExecPlan
+            MP ->> MP: rhs: ExecPlan
+       end
+    end
+    MP -->> MP: createStitchExecPlan(InProcessDispatcher)
+    MP -->> QE: StitchExecPlan
+    QE -->> C: StitchExecPlan
+    C -->> C: execute stitch plan in process
```

Comment:
We need to discuss more about how to document this step more clearly here. I know that you have the idea right.

Reply:
Update the sequence diagram with more details around InProcessPlanDispatcher.

Reply:
Latest sequence diagram [link](https://mermaidjs.github.io/mermaid-live-editor/#/view/eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG4gICAgcGFydGljaXBhbnQgQyBhcyBDbGllbnRcbiAgICBwYXJ0aWNpcGFudCBRRSBhcyBRdWVyeUVuZ2luZVtjbHVzdGVyMV1cbiAgICBwYXJ0aWNpcGFudCBGRDEgYXMgRmlsb0RCW2NsdXN0ZXIxXVxuICAgIHBhcnRpY2lwYW50IElQRCBhcyBjbGFzcyBJblByb2Nlc3NQbGFuRGlzcGF0Y2hlclxuICAgIHBhcnRpY2lwYW50IFJQRCBhcyBjbGFzcyBSZW1vdGVQbGFuRGlzcGF0Y2hlclxuICAgIHBhcnRpY2lwYW50IFFFMiBhcyBRdWVyeUVuZ2luZVtjbHVzdGVyMl1cbiAgICBwYXJ0aWNpcGFudCBNUCBhcyBkZWYgbWF0ZXJpYWxpemVQbGFuXG4gICAgcGFydGljaXBhbnQgRlAgYXMgdHJhaXQgRmFpbHVyZVByb3ZpZGVyXG4gICAgcGFydGljaXBhbnQgUlAgYXMgdHJhaXQgUm91dGluZ1BsYW5uZXJcbiAgICBDLT4-UUU6IFBhc2VyLnF1ZXJ5VG9Mb2dpY2FsUGxhblxuICAgIFFFLS0-PkM6IExvZ2ljYWxQbGFuXG4gICAgQy0-PlFFOiBtYXRlcmlhbGl6ZVBsYW5cbiAgICBRRS0-Pk1QOiBtYXRlcmlhbGl6ZVBsYW5cbiAgICBNUC0-PkZQOiBnZXRGYWlsdXJlc1xuICAgIEZQLS0-Pk1QOiBTZXFbRmFpbHVyZVRpbWVSYW5nZV1cbiAgICBNUC0-PlJQOiBwbGFuKFNlcVtGYWlsdXJlVGltZVJhbmdlXSlcbiAgICBSUC0tPj5NUDogU2VxW1JvdXRlXVxuICAgIGxvb3AgZm9yIGVhY2ggUm91dGVcbiAgICAgICAgYWx0IGlzIExvY2FsUm91dGVcbiAgICAgICAgICAgIE1QIC0-PiBNUDogdXBkYXRlTG9naWNhbFBsYW5XaXRoVGltZVJhbmdlXG4gICAgICAgICAgICBNUCAtPj4gTVA6IHdhbGtMb2dpY2FsUGxhblRyZWVcbiAgICAgICAgICAgIE1QIC0-PiBNUDogbGhzOiBFeGVjUGxhblxuICAgICAgICBlbHNlIGlzIFJlbW90ZVJvdXRlXG4gICAgICAgICAgICBNUCAtPj4gTVA6IHVwZGF0ZUxvZ2ljYWxQbGFuV2l0aFRpbWVSYW5nZVxuICAgICAgICAgICAgTVAgLT4-IE1QOiBjcmVhdGVSZW1vdGVFeGVjUGxhblxuICAgICAgICAgICAgTVAgLT4-IE1QOiByaHM6IEV4ZWNQbGFuXG4gICAgICAgZW5kXG4gICAgZW5kXG4gICAgTVAgLS0-PiBNUDogY3JlYXRlU3RpdGNoRXhlY1BsYW4oSW5Qcm9jZXNzRGlzcGF0Y2hlcilcbiAgICBNUCAtLT4-IFFFOiBTdGl0Y2hFeGVjUGxhblxuICAgIFFFIC0tPj4gQzogU3RpdGNoRXhlY1BsYW5cbiAgICBDIC0-PiBRRTogU3RpdGNoRXhlY1BsYW4uZGlzcGF0Y2hcbiAgICBRRSAtPj4gSVBEOiBkaXNwYXRjaFxuICAgIElQRCAtLT4-IElQRDogRXhlY1BsYW4uZXhlY3V0ZShEdW1teUNodW5rU291cmNlKVxuICAgIGFsdCBpcyBMb2NhbCBFeGVjUGxhblxuICAgICAgIElQRCAtPj4gRkQxOiBkaXNwYXRjaCBhbmQgZXhlY3V0ZSBFeGVjUGxhblxuICAgICAgIEZEMSAtLT4-IElQRDogUXVlcnlSZXNwb25zZVxuICAgIGVsc2UgaXMgUmVtb3RlIEV4ZWNQbGFuXG4gICAgICAgSVBEIC0-PiBSUEQ6IGRpc3BhdGNoXG4gICAgICAgUlBEIC0-PiBRRTI6IG1hdGVyaWFsaXplIGFuZCBleGVjdXRlIExvZ2ljYWxQbGFuXG4gICAgICAgUUUyIC0tPj4gUlBEOiBRdWVyeVJlc3BvbnNlXG4gICAgICAgUlBEIC0tPj4gSVBEOiBRdWVyeVJlc3BvbnNlXG4gICAgZW5kXG4gICAgSVBEIC0tPj4gSVBEOiAgU3RpdGNoIFF1ZXJ5UmVzcG9uc2VcbiAgICBJUEQgLS0-PiBRRTogUXVlcnlSZXNwb25zZVxuICAgIFFFIC0tPj4gQzogUXVlcnlSZXNwb25zZVxuIiwibWVybWFpZCI6eyJ0aGVtZSI6ImRlZmF1bHQifX0)